torch/_dynamo/variables/builtin.py

```diff
@@ -558,7 +553,11 @@ def call_function(
                         args[0], variables.UnspecializedPythonVariable
                     ):
                         args[0] = args[0].convert_to_constant(tx)
-                    return wrap_fx_proxy(tx, proxy, **options)
+                    if check_numpy_ndarray_args(args, kwargs):
+                        cls = variables.NumpyNdarrayVariable
+                    else:
+                        cls = variables.TensorVariable
+                    return wrap_fx_proxy_cls(cls, tx, proxy, **options)
```

Comment:
This just seems like a refactor? Or is this also fixing the guard problem?

Reply:
It makes the tensor and ndarray path the same, so that they both behave the same. For now, this just means the behaviour wrt. `truediv`, but we don't want them to diverge further in the future.